code/datums/traits/good.dm

```diff
@@ -96,6 +96,28 @@
 	)
 	H.equip_in_one_of_slots(B, slots , qdel_on_fail = TRUE)
 
+/datum/quirk/multilingual
+	name = "Multilingual"
+	desc = "You spent a portion of your life learning to understand an additional language. You may or may not be able to speak it based on your anatomy."
+	value = 2
+	mob_trait = TRAIT_MULTILINGUAL
+	gain_text = "<span class='notice'>You have learned to understand an additional language.</span>"
+	lose_text = "<span class='danger'>You have forgotten how to understand a language.</span>"
+
+/datum/quirk/multilingual/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	if(H.job != "Curator")
+		var/obj/item/organ/tongue/T = H.getorganslot(ORGAN_SLOT_TONGUE)
+		var/list/languages_possible = T.languages_possible
+		languages_possible = languages_possible - typecacheof(/datum/language/common) - typecacheof(/datum/language/uncommon) - typecacheof(/datum/language/codespeak) - typecacheof(/datum/language/narsie) - typecacheof(/datum/language/ratvar)
```

Comment:
🤮

Reply:
Name any better method!

Reply:
It was either this, or to play with the LANGUAGE_MULTILINGUAL source to block those languages, grant all languages, pick a random language, remove all languages, unblock those languages, and then add the random language.

I'd take this over code getting fucky sometimes and absolutely ruining the trait 1/1000 times.

Reply:
Although this one still has dumb random loop 2 lines down, it's just a very astronomically small chance it would fail for more than a few loops... but in theory it could get stuck there forever 😄